### PR TITLE
Docs: Add snapshot testing doc

### PR DESCRIPTION
**Pull request by sirreal**
This PR adds a snapshot testing doc as suggested by @gziolo. It is largely based on the [Calypso snapshot testing doc](https://github.com/Automattic/wp-calypso/blob/b33c698524471eaed63051223f310a87030e9336/docs/testing/snapshot-testing.md).

I've done a pass to try to ensure the information is accurate for Gutenberg, but it would be great to get some feedback from folks more familiar with the project. It would also be great to add links in the appropriate places.

### CODE REVIEW

**Comment by gziolo** [comment]
Thanks @sirreal for opening this PR and making it Gutenberg specific. Awesome work 💯

I made it more general so we could link it next to Coding Guidelines. I think it's a good start and we should merge it as it is. We will have to add further refinements to cover unit, component testing and e2e testing.

**Comment by youknowriad** [comment]
This looks good to me, I think we should also move our current Testing docs (in the CONTRIBUTING.md) file to this newly created doc for consistency.

**Review by BE-Webdesign — APPROVED** [review]
Looks great. Thank you.

**Comment by gziolo** [comment]
> This looks good to me, I think we should also move our current Testing docs (in the CONTRIBUTING.md) file to this newly created doc for consistency.

@youknowriad, great idea. I will take care of it on Friday 💯

**Comment by sirreal** [comment]
🙇 Thanks everyone!

**Comment by pento** [comment]
Because we use Jetpack Markdown, the docs files can't include line breaks - the Markdown spec allows them, but Jetpack converts them to `<br/>`, for consistency with WordPress' behaviour.

File: docs/testing-overview.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+```
+
+Reducer tests are also be a great fit for snapshots. They are often large, complex data structures
+that shouldn't change unexpectedly, exactly what snapshots excel at!
+
+### Working with snapshots
+
+You might be blindsided by CI tests failing when snapshots don't match. You'll need to
+[update snapshots] if the changes are expected. The quick and dirty solution is to invoke Jest with
+`--updateSnapshot`. In Calypso you can do that as follows:
```
Comment (aduth) [inline docs/testing-overview.md]: We're not "In Calypso" 😬

Comment (sirreal) [reply docs/testing-overview.md]: 😬 Thanks for spotting that… PR incoming.

Comment (gziolo) [reply docs/testing-overview.md]: 😎 we have one more PR open with docs ported from Calypso. I’ll make sure we don’t repeat the same mistake 😃

**Pull request merged**